pkg/controller/controller.go

```diff
@@ -90,9 +95,35 @@ func (c *Controller) Run(ctx context.Context, cacheRefreshRate time.Duration) er
 		DeleteFunc: c.deleteObject,
 	})
 
+	nodeInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			// Add node info and data to the map
+			err := c.nodesArchitecture.AddNode(obj.(*corev1.Node))
+			if err != nil {
+				// we could log the error
+				return
+			}
+		},
+		UpdateFunc: func(old, new interface{}) {
+			// override the map
+			err := c.nodesArchitecture.AddNode(new.(*corev1.Node))
+			c.log.Info("updating node from node informer")
+			if err != nil {
+				return
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			// remove node from the map
+			err := c.nodesArchitecture.DeleteNode((obj.(*corev1.Node)).Name)
+			if err != nil {
+				return
+			}
+		},
+	})
+
```

Comment:
Move into /pkg/controller/nodes

Reply:
Just the node informer? So the `/pkg/controller/nodes` will just be a wrapper around the informer